File: app/src/main/java/com/gmail/fattazzo/formula1world/domain/F1FastestLap.java
package com.gmail.fattazzo.formula1world.domain;

/**
 * @author fattazzo
 *         <p/>
 *         date: 06/06/17
 */
public class F1FastestLap {

    public int rank;

    public int lap;

    public F1Time time;

    public F1AverageSpeed averageSpeed;

    public F1FastestLap() {
    }

    public F1FastestLap(int rank, int lap, F1Time time, F1AverageSpeed averageSpeed) {
        this.rank = rank;
        this.lap = lap;
        this.time = time;
        this.averageSpeed = averageSpeed;
    }

    @Override
    public String toString() {
        return "FastestLap{" +
                "rank=" + rank +
                ", lap=" + lap +
                ", time=" + time +
                ", averageSpeed=" + averageSpeed +
                '}';
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/domain/F1FastestLap.kt
package com.gmail.fattazzo.formula1world.domain

/**
 * @author fattazzo
 *
 *
 * date: 06/06/17
 */
class F1FastestLap {

    var rank: Int = 0

    var lap: Int = 0

    var time: F1Time? = null

    var averageSpeed: F1AverageSpeed? = null

    constructor() {}

    constructor(rank: Int, lap: Int, time: F1Time, averageSpeed: F1AverageSpeed) {
        this.rank = rank
        this.lap = lap
        this.time = time
        this.averageSpeed = averageSpeed
    }

    override fun toString(): String {
        return "FastestLap{" +
                "rank=" + rank +
                ", lap=" + lap +
                ", time=" + time +
                ", averageSpeed=" + averageSpeed +
                '}'
    }
}
